src/main/java/org/jabref/gui/mergeentries/FetchAndMergeEntry.java

```diff
@@ -41,11 +54,41 @@ public FetchAndMergeEntry(BibEntry entry, String field) {
      * @param fields - List of fields to get information from, one at a time in given order
      */
     public FetchAndMergeEntry(BibEntry entry, BasePanel panel, List<String> fields) {
+        this.dialogService = panel.frame().getDialogService();
+        this.panel = panel;
+
+        // TODO: Don't run this method as part of the constructor
+        fetchAndMerge(entry, fields);
+    }
+
+    public void fetchAndMerge(BibEntry entry, List<String> fields) {
         for (String field : fields) {
-            if (entry.hasField(field)) {
-                new FetchAndMergeWorker(panel, entry, field).execute();
+            Optional<String> fieldContent = entry.getField(field);
+            if (fieldContent.isPresent()) {
+                BackgroundTask.wrap(() -> {
+                    Optional<IdBasedFetcher> fetcher = WebFetchers.getIdBasedFetcherForField(field, Globals.prefs.getImportFormatPreferences());
+                    if (fetcher.isPresent()) {
+                        return fetcher.get().performSearchById(fieldContent.get());
+                    } else {
+                        return Optional.<BibEntry>empty();
```

Comment:
Normally you don't need the genericc type in return Optional.empty

Reply:
I was also surprised but it leads to compiler errors without it.